pkg/integrations/cadvisor/cadvisor.go

```diff
@@ -52,30 +53,27 @@ var disabledMetricsSet = container.MetricSet{
 
 // GetIncludedMetrics applies some logic to determine the final set of metrics to be scraped and returned by the cAdvisor integration
 func (c *Config) GetIncludedMetrics() (container.MetricSet, error) {
-	var enabledMetrics, includeMetrics container.MetricSet
-	// Clear default disabled metrics if explicit disabled metrics are configured
-	if len(c.DisabledMetrics) > 0 {
-		disabledMetricsSet = container.MetricSet{}
-	}
-	for _, d := range c.DisabledMetrics {
-		if err := disabledMetricsSet.Set(d); err != nil {
-			return includeMetrics, fmt.Errorf("failed to set disabled metric: %w", err)
+	var enabledMetrics, includedMetrics container.MetricSet
+
+	if c.DisabledMetrics != nil {
```

Comment:
Both `c.DisabledMetrics` and `c.EnabledMetrics` have zero values that come from the `Config` struct , so these conditionals would always pass, and set the enabled. and disabled sets to empty lists.

https://go.dev/play/p/Brw6Dc27WHn

Reply:
The zero value of a slice is `nil`. It can be a bit confusing because printing a `nil` slice results in `[]`.

https://github.com/golang/go/issues/12780
https://go.dev/play/p/Q_PaZe6c3Go

Reply:
TIL. Thanks for setting me straight on this!